src/lua/funcs/lua_registry.rs

```diff
@@ -0,0 +1,131 @@
+//! Metamethods for accesssing the registry values!
+
+use super::super::super::registry;
+use registry::RegistryAccess;
+
+use hlua::{LuaTable};
+use hlua::any::AnyLuaValue;
+
+use rustc_serialize::json::{Json, ToJson};
+use std::ops::Deref;
+pub fn index(table: AnyLuaValue, lua_key: AnyLuaValue)
+             -> Result<AnyLuaValue, &'static str> {
+    if let AnyLuaValue::LuaString(key) = lua_key {
```

Comment:
Is the reason we need this check (instead of encoding this in the function signature) because this is a function that will be called from lua and so it can be any lua value?